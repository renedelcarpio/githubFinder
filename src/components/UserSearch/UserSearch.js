import { useState } from 'react';
import { User } from '../../services/services';
import ErrorMessage from '../ErrorMessage/ErrorMessage';
import Form from '../Form/Form';
import Loader from '../Loader/Loader';
import UserCard from '../UserCard/UserCard';

import './UserSearch.scss';

const UserSearch = () => {
  const [user, setUser] = useState('');
  const [data, setData] = useState([]);
  const [loader, setLoader] = useState(false);
  const [errorMessage, setErrorMessage] = useState('');

  const onUserChange = (e) => setUser(e.target.value);

  const getUsers = async () => {
    setLoader(true);
    try {
      await User.getUser(user).then(({ items }) => {
        setData(items);
        setLoader(false);
      });
    } catch (error) {
      console.log(error);
      setErrorMessage(`${error.message}, the user does not exist.`);
      setLoader(false);
    }
  };

  const onSearchUser = (e) => {
    e.preventDefault();
    setErrorMessage('');
    setData([]);
    getUsers();
  };

  return (
    <div className="search__container">
      <Form onSubmit={onSearchUser} value={user} onChange={onUserChange} />
      {loader ? (
        <Loader />
      ) : (
        <div className="user">
          {errorMessage ? (
            <ErrorMessage message={errorMessage} />
          ) : (
            data?.map((person) => {
              return (
                <UserCard
                  key={person.id}
                  image={person.avatar_url}
                  name={person.login}
                  url={person.html_url}
                />
              );
            })
          )}
        </div>
      )}
    </div>
  );
};

export default UserSearch;
