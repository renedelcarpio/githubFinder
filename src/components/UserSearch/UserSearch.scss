@import '../../index.scss';

.search__container {
  display: flex;
  flex-direction: column;
  align-items: center;
  background: $background-dark;
  width: 100vw;
  min-height: 100vh;
}

.user {
  display: flex;
  flex-wrap: wrap;
  width: 80%;
}

@media (max-width: 770px) {
  .user {
    width: 95%;
  }
}
