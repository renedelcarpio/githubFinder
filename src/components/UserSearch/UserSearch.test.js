import { shallow } from 'enzyme';
import UserSearch from './UserSearch';

const wrapper = shallow(<UserSearch />);

describe('tests in <UserSearch/> component', () => {
  test('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
