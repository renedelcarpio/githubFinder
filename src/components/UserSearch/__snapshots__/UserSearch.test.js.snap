// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in <UserSearch/> component should match the snapshot 1`] = `
<div
  className="search__container"
>
  <Form
    onChange={[Function]}
    onSubmit={[Function]}
    value=""
  />
  <div
    className="user"
  />
</div>
`;
